### ISSUE TITLE
improved error messages for setExpr when the proper fields are not provided 

### ISSUE THREAD

**Issue by leanmiguel** [guregu/dynamo#212]
recently had a doh moment with setExpr because I forgot to add the fields required to populate the expression. 
perhaps an improved error message could be used here? 

example incorrect setExpr
```
If("attribute_exists('foo'.$)") // nothing provided to substitute the $
```
example error
```
{
	"message": "runtime error: index out of range [0] with length 0"
}
```

**Comment by guregu** [comment]
Thanks, we should definitely return a better error message here and avoid the bad slice access.

**Comment by guregu** [comment]
Just released a fix in [v1.18.2](https://github.com/guregu/dynamo/tree/v1.18.2), thanks!
